$OpenBSD: patch-kipi-plugins_kameraklient_gpcamera_cpp,v 1.1 2013/01/19 11:27:01 zhuk Exp $
Fix compilation with GPhoto 2.5. Based on ideas from this patch:
http://bugsfiles.kde.org/attachment.cgi?id=73176
--- kipi-plugins/kameraklient/gpcamera.cpp.orig	Sat Jan 19 00:43:53 2013
+++ kipi-plugins/kameraklient/gpcamera.cpp	Sat Jan 19 00:46:03 2013
@@ -436,7 +436,7 @@ int GPCamera::uploadItem(const QString& folder, const 
         status = 0;
     }
     status = new GPStatus;
-    if (gp_camera_folder_put_file(d->camera, folder.latin1(), cfile, status->context) != GP_OK) {
+    if (gp_camera_folder_put_file(d->camera, folder.latin1(), itemName.latin1(), GP_FILE_TYPE_NORMAL, cfile, status->context) != GP_OK) {
         gp_file_unref(cfile);
         delete status;
         status = 0;
@@ -530,7 +530,9 @@ void GPCamera::getSupportedPorts(QStringList& plist) {
     int numPorts = gp_port_info_list_count( list );
     for (int i = 0; i < numPorts; i++) {
         gp_port_info_list_get_info( list, i, &info );
-        plist.append(info.path);
+        char *path;
+        gp_port_info_get_name( info, &path );
+        plist.append(path);
     }
     gp_port_info_list_free( list );
 }
